### ISSUE TITLE
feature: new rule multiline-if-init

### ISSUE THREAD

**Issue by marcopeereboom** [mgechev/revive#1714]
The `if`-init idiom in Go is designed for tight one-liners like `if err := foo(); err != nil`. When the init clause spans multiple lines — typically because it contains a struct literal or a wrapped call chain — the boundary between initialization and condition becomes hard to see:

```go
if err := insert(&MempoolTx{
    id: tx.TxHash(), expires: time.Now().Add(time.Hour), tx: tx,
}); err != nil {
    t.Fatal(err)
}
```

The `});` closing is syntactically valid but visually deceptive. This pattern comes up frequently in code review and the fix is always the same: extract the init to a separate statement.

**Describe the solution you'd like**

A new opt-in rule `multiline-if-init` that flags `if` statements whose init clause spans more than one line. The rule walks `ast.IfStmt` nodes and compares the start line of `Init.Pos()` with the end line of `Init.End()`. If they differ, it reports a failure.

The fix is mechanical:

```go
err := insert(&MempoolTx{
    id: tx.TxHash(), expires: time.Now().Add(time.Hour), tx: tx,
})
if err != nil {
    t.Fatal(err)
}
```

No configuration needed. No type checking required. No false positives on legitimate single-line multi-return inits like `if x, err := foo(); err != nil`.

**Describe alternatives you've considered**

- Enforcing this purely through code review, which is how we've been doing it. It works but it's tedious and error-prone — reviewers have to catch it manually every time.
- A project-local linter plugin, but revive's `-buildmode=plugin` support is limited and this rule is general enough to benefit the wider Go community.

**Additional context**

I have a working implementation with tests ready to submit as a PR. Ran it against a ~100k line Go codebase and it found 24 violations with zero false positives.

**Comment by marcopeereboom** [comment]
bump
